velox/experimental/cudf/connectors/hive/CudfHiveConfig.h

```diff
@@ -81,6 +81,12 @@ class CudfHiveConfig {
   static constexpr const char* kTimestampTypeSession =
       "parquet.reader.timestamp_type";
 
+  // Whether to use async data cache aided buffered input source
+  static constexpr const char* kUseBufferedInput =
```

Comment:
Cudf supports many file formats, may need to separate the config for common CudfConfig and file format specified config in the future.

Reply:
>Cudf supports many file formats, may need to separate the config for common CudfConfig and file format specified config in the future.

Updated the config option to `cudf.hive.use-buffered-input` and updated comments

Reply:
> `cudf.hive.async-data-cache-enabled`

This is basically a `use_velox_io`. it can be used with or without async data cache. although you need it turned on for async data cache to take effect in cudf's readers. you still also need to enable async data cache.

Reply:
> Let's use cudf.hive.async-data-cache-enabled.

and

> The AsyncDataCache currently has to be initialized via setInstance(). That will control enabling/disabling the cache.
http://github.com/facebookincubator/velox/blob/main/velox/common/caching/AsyncDataCache.cpp#L700

I think `cudf.hive.use-buffered-input` is the correct option here. The file handle AsyncDataCache is initialized in `fileHandleFactory` of the HiveConnector at construction based on `HiveConfig:: kEnableFileHandleCache` config (default: on). Whereas in `CudfHiveDataSourece::createSplitReader()` we are only deciding if we want to use the buffered input data source (which may benefit from the file handle cache if available and enabled, and would still work without the cache).

Reply:
>This is basically a use_velox_io. it can be used with or without async data cache. although you need it turned on for async data cache to take effect in cudf's readers. you still also need to enable async data cache.

This is okay too. Pretty flexible on the names here as long we are on the same page 😄

Reply:
> Cudf supports many file formats, may need to separate the config for common CudfConfig and file format specified config in the future.

xref: https://github.com/facebookincubator/velox/pull/15325/files#r2480784073